client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,152 +1,194 @@
+        this.types = [
+            { value: "internal", label: this.$translate.instant("telephony_alias_configuration_agents_type_internal") },
+            { value: "external", label: this.$translate.instant("telephony_alias_configuration_agents_type_external") }
+        ];
+
+        this.onChooseServicePopover = ({ serviceName }, pos) => {
+            this.addAgentForm.numbers[pos] = serviceName;
+        };
+
+        return this.fetchAgentsIds().then((ids) => {
+            this.agents.ids = ids;
+        }).catch((err) => {
+            this.Toast.error([this.$translate.instant("telephony_alias_configuration_agents_get_error"), _.get(err, "data.message")].join(" "));
+            return this.$q.reject(err);
         });
     }
 
-    self.fetchAgentsIds = function () {
-        self.agents.isLoading = true;
-        return OvhApiTelephony.OvhPabx().Hunting().Agent().v6().query({
-            billingAccount: $stateParams.billingAccount,
-            serviceName: $stateParams.serviceName
-        }).$promise.finally(function () {
-            self.agents.isLoading = false;
+    orderBy (prop, asc) {
+        this.orderedAscAgents = !this.orderedAscAgents;
+        if (asc != null) {
+            this.orderedAscAgents = asc;
+        }
+
+        this.agents.paginated = _.sortBy(this.agents.paginated, prop);
+
+        if (!this.orderedAscAgents) {
+            this.agents.paginated = this.agents.paginated.reverse();
+        }
+    }
+
+    fetchAgentsIds () {
+        this.agents.isLoading = true;
+        return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6().query({
+            billingAccount: this.$stateParams.billingAccount,
+            serviceName: this.$stateParams.serviceName
+        }).$promise.finally(() => {
+            this.agents.isLoading = false;
         });
-    };
+    }
 
-    self.fetchAgent = function (id) {
-        return OvhApiTelephony.OvhPabx().Hunting().Agent().v6().get({
-            billingAccount: $stateParams.billingAccount,
-            serviceName: $stateParams.serviceName,
+    fetchAgent (id) {
+        return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6().get({
+            billingAccount: this.$stateParams.billingAccount,
+            serviceName: this.$stateParams.serviceName,
             agentId: id
         }).$promise;
-    };
-
-    self.getSelectedAgentIds = function () {
-        return _.keys(self.agents.selected);
-    };
-
-    self.deleteAgents = function () {
-        self.agents.isDeleting = true;
-        return $q.all(self.getSelectedAgentIds().map(function (id) {
-            return OvhApiTelephony.OvhPabx().Hunting().Agent().v6().remove({
-                billingAccount: $stateParams.billingAccount,
-                serviceName: $stateParams.serviceName,
+    }
+
+    getSelectedAgentIds () {
+        return _.keys(this.agents.selected);
+    }
+
+    onTransformItemDone (items) {
+        this.$timeout(() => this.orderBy("number", this.orderedAscAgents), 1);
```

Comment:
why using a `$timeout` here? I guess it's for forcing a new digest cycle. If so you can omit the `delay` param.